urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/core/Tablero.java

```diff
@@ -0,0 +1,183 @@
+package core;
+
+import java.awt.Point;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import logica.ExploradorMovimientos;
+
+public class Tablero {
+	
+	public static int TABLERO_LARGO = 10;
+	public static int TABLERO_ANCHO = 10;
+	
+	private Map<Point, EstadoCasilla> tablero;
+	private static int casillasjugablesIniciales;
+	
+	public enum TipoTablero {
+		CLASICO,
+		OCTOGONAL,
+		PERSONALIZADO;
+	}
+	private TipoTablero tipoTablero;
+	
+	public Tablero() {
+		tablero = new HashMap<Point, EstadoCasilla>(TABLERO_LARGO * TABLERO_ANCHO);
+		this.tipoTablero = TipoTablero.CLASICO;
+		setCasillasjugablesIniciales(0);
+		inicializar();
+	}
+	
+	public Tablero(int largo, int ancho, TipoTablero tipoTablero) {
+		TABLERO_LARGO = largo;
+		TABLERO_ANCHO = ancho;
+		this.tipoTablero = tipoTablero;
+		setCasillasjugablesIniciales(0);
+		tablero = new HashMap<Point, EstadoCasilla>(TABLERO_LARGO * TABLERO_ANCHO);
+		inicializar();
+	}
+	
+	public Tablero(Map<Point, EstadoCasilla> tablero, TipoTablero tipoTablero) {
+		super();
+		this.tipoTablero = tipoTablero;
+		setCasillasjugablesIniciales(0);
+		if (tipoTablero == null) {
+			this.tipoTablero = TipoTablero.PERSONALIZADO;
+		}
+		this.tablero = new HashMap<Point, EstadoCasilla>(tablero.size());
+		for(Point point : tablero.keySet()) {
+			this.tablero.put(new Point(point), tablero.get(point));
+			if(tablero.get(point) == EstadoCasilla.EMPTY){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.